transport/src/main/java/io/netty/channel/DefaultChannelPromiseAggregatorFactory.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import io.netty.util.concurrent.EventExecutor;
+
+/**
+ * Provides the ability to associate the outcome of multiple {@link ChannelPromise}
+ * objects into a single {@link ChannelPromise} object.
+ * <p>
+ * Not thread safe. Must be used within the {@link EventLoop}.
+ */
+public final class DefaultChannelPromiseAggregatorFactory extends DefaultChannelPromise
+                                                          implements ChannelPromiseAggregatorFactory {
+    private final ChannelPromise promise;
+    private int expectedCount;
+    private int doneCount;
+    private Throwable lastFailure;
+    private boolean doneAllocating;
+
+    /**
+     * Create a new instance which returns an appropriate type if {@code aggregationResult} is {@link #isVoid()}.
+     * @param aggregationResult The promise whose status will reflect the result of the aggregation.
+     * @param executor the {@link EventExecutor} which is used to notify the promise once it is complete.
+     */
+    public static ChannelPromiseAggregatorFactory newInstance(ChannelPromise aggregationResult,
+                                                              EventExecutor executor) {
+        return aggregationResult.isVoid() ? new VoidChannelPromiseAggregatorFactory(aggregationResult.channel(), true)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Ternary operators that can be simplified with || or && [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3ASimplifiedTernary)